src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -1138,6 +1154,60 @@ private MethodSymbol PerformForEachPatternOverloadResolution(TypeSymbol patternT
             return result;
         }
 
+        private MethodSymbol FindForEachPatternMethodViaExtension(BoundExpression collectionExpr, string methodName, DiagnosticBag diagnostics)
+        {
+            var analyzedArguments = AnalyzedArguments.GetInstance();
+
+            var methodGroupResolutionResult = this.BindExtensionMethod(
+                _syntax.Expression,
+                methodName,
+                analyzedArguments,
+                collectionExpr,
+                typeArgumentsWithAnnotations: default,
+                isMethodGroupConversion: false,
+                returnRefKind: default,
+                returnType: null);
+
+            diagnostics.AddRange(methodGroupResolutionResult.Diagnostics);
+
+            var overloadResolutionResult = methodGroupResolutionResult.OverloadResolutionResult;
+            if (overloadResolutionResult?.Succeeded ?? false)
+            {
+                var result = overloadResolutionResult.ValidResult.Member;
+
+                if (result.CallsAreOmitted(_syntax.SyntaxTree))
+                {
+                    // Calls to this method are omitted in the current syntax tree, i.e it is either a partial method with no implementation part OR a conditional method whose condition is not true in this source file.
+                    // We don't want to want to allow this case, see StatementBinder::bindPatternToMethod.
```

Comment:
>StatementBinder::bindPatternToMethod [](start = 69, length = 36)

Not sure what this reference is referring to.

Reply:
That's copied straight from https://github.com/dotnet/roslyn/blob/29542b0a669071b34fc84fe679c767409460b34f/src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs#L1063-L1068

I have no idea what it's referring to either. I can't find any such method in roslyn. I'll just delete that part of the comment.

Reply:
Generally something from the native compiler that was carried over during porting.